internal/core/window.rs

```diff
@@ -363,15 +378,27 @@ impl WindowInner {
     /// Arguments:
     /// * `event`: The key event received by the windowing system.
     /// * `component`: The Slint compiled component that provides the tree of items.
-    pub fn process_key_input(&self, event: &KeyEvent) {
+    pub fn process_key_input(&self, event: KeyInputEvent) {
+        if let Some(updated_modifier) = self
+            .modifiers
+            .get()
+            .state_update(event.event_type == KeyEventType::KeyPressed, &event.text)
+        {
+            // Updates the key modifiers depending on the key code and pressed state.
+            self.modifiers.set(updated_modifier);
+            return;
```

Comment:
I think we should still forward the event for that key.
```suggestion
```

Reply:
That is what breaks existing tests. So for example

```
                key-pressed(event) => {
                    received += event.text;
                    accept
                }
```

in `event_propagation.slint` would become

```
                key-pressed(event) => {
                    if (!event.control && !event.shift && ...) {
                        received += event.text;
                    }
                    accept
                }
```

... or some variant of that.

That would also be a breaking change.

I don't think that we should do it (hence I wrote the `return` statement). I'd rather have a `modifiers-changed` callback in `FocusScope`.

Reply:
I think the test need to be adapted. It's IMHO the right behavior to have event for these keys.
Say i want to do something when the control key is pressed.

> That would also be a breaking change.

I think it's acceptable since it is just adding new key.
You already need to filter for key you don't understand in your key event.

Reply:
But the point is that i can't think of a way that the code would break. 
You already need to filter special keys otherwise you get bugs when user press the arrows or PrintScreen or such keys.
Not a worse breakage than ctrl <-> cmd mapping.

In other words, i'm saying this is not a breaking change.

Reply:
Ok, I'll change it.